uaclient/tests/test_cli_attach_premium.py

```diff
@@ -24,9 +28,118 @@ def test_non_root_users_are_rejected(getuid):
         action_attach_premium(mock.MagicMock(), cfg)
 
 
+class TestGetContractTokenFromCloudIdentity:
+    @pytest.mark.parametrize("cloud_type", ("awslookalike", "azure", "!aws"))
+    @mock.patch("uaclient.clouds.identity.get_cloud_type")
+    def test_non_aws_cloud_type_returns_no_token(
+        self, get_cloud_type, cloud_type, capsys
+    ):
+        """Non-aws clouds will not return a token."""
+        get_cloud_type.return_value = cloud_type
+        assert None is _get_contract_token_from_cloud_identity(FakeConfig())
+        out = capsys.readouterr()[0]
```

Comment:
A nit: we do `out, _err = capsys.readouterr()` almost everywhere that we only want `out`, it would be good to be consistent. (This applies throughout these new tests, I think.)

Reply:
+1